nixos/modules/config/stub-ld.nix

```
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) last splitString mkOption types mdDoc mkIf optionals;
+
+  message = ''
+    You seem to be trying to run a dynamically linked executable that
+    wasn't made for NixOS. This is possible, but not without extra steps,
+    because NixOS, by design, does not have a global library path.
```

I would as discussed to cut this down to one paragraph:

```suggestion
    You seem to be trying to run a dynamically linked executable that
    wasn't made for NixOS. This is not supported without extra steps.
    See this link for further information: <link>
```

Reason, if you have a program going riot this will be printed potentially many times in a row.

---

The larger text could be put into the manual instead and replaced with a link.

All this program really needs to output is:

    Running an FHS binaries on NixOS is not supported. Please see <link> for more information.

---

@Atemu fhs binary is a bit too much NixOS lingo, I think. People getting started with NixOS might have a better understanding what a dynamically linked executable is.

---

Yeah, it could get really spammy in some cases. Though, how common would that be?

I do think having actual solutions at least summarized in the output is pretty valuable, though the details of point 2 are definitely better left online.